infra/main.bicep

```diff
@@ -0,0 +1,746 @@
+          version: '2'
+        }
+        capacity: 30
+      }
+    ]
+  }
+}
+
+module speechService 'core/ai/cognitiveservices.bicep' = {
+  scope: rg
+  name: speechServiceName
+  params:{
+    name: speechServiceName
+    location: location
+    sku: {
+      name: 'S0'
+    }
+    kind: 'SpeechServices'
+  }
+}
+
+module storekeys './app/storekeys.bicep' = if (useKeyVault) {
+  name: 'storekeys'
+  scope: rg
+  params: {
+    keyVaultName: keyVaultName
+    azureOpenAIName: openai.outputs.name
+    azureCognitiveSearchName: search.outputs.name
+    storageAccountName: storage.outputs.name
+    formRecognizerName: formrecognizer.outputs.name
+    contentSafetyName: contentsafety.outputs.name
+    speechServiceName: speechServiceName
+    rgName: rgName
+  }
+}
+
+module search './core/search/search-services.bicep' = {
+  name: azureCognitiveSearchName
+  scope: rg
+  params:{
+    name: azureCognitiveSearchName
+    location: location
+    tags: { 
+      deployment : searchTag
+    }
+    sku: {
+      name: azureCognitiveSearchSku
+    }
+    authOptions: {
+      aadOrApiKey: {
+        aadAuthFailureMode: 'http403'
+      }
+    }
+  }
+}
+
+module hostingplan './core/host/appserviceplan.bicep' = {
+  name: hostingPlanName
+  scope: rg
+  params: {
+    name: hostingPlanName
+    location: location
+    sku: {
+      name: hostingPlanSku
+    }
+    reserved: true
+  }
+}
+
+module web './app/web.bicep' = {
+  name: websiteName
+  scope: rg
+  params: {
+    name: websiteName
+    location: location
+    tags: { 'azd-service-name': 'web' }
+    appServicePlanId: hostingplan.outputs.name
+    applicationInsightsName: monitoring.outputs.applicationInsightsName
+    azureOpenAIName: openai.outputs.name
+    azureCognitiveSearchName: search.outputs.name
+    storageAccountName: storage.outputs.name
+    formRecognizerName: formrecognizer.outputs.name
+    contentSafetyName: contentsafety.outputs.name
+    speechServiceName: speechService.outputs.name
+    openAIKeyName: useKeyVault ? storekeys.outputs.OPENAI_KEY_NAME : ''
+    storageAccountKeyName: useKeyVault ? storekeys.outputs.STORAGE_ACCOUNT_KEY_NAME : ''
+    formRecognizerKeyName: useKeyVault ? storekeys.outputs.FORM_RECOGNIZER_KEY_NAME : ''
+    searchKeyName: useKeyVault ? storekeys.outputs.SEARCH_KEY_NAME : ''
+    contentSafetyKeyName: useKeyVault ? storekeys.outputs.CONTENT_SAFETY_KEY_NAME : ''
+    speechKeyName: useKeyVault ? storekeys.outputs.SPEECH_KEY_NAME: ''
+    useKeyVault: useKeyVault
+    keyVaultName: useKeyVault || authType == 'rbac' ? keyvault.outputs.name : ''
+    keyVaultEndpoint: useKeyVault ? keyvault.outputs.endpoint : ''
+    authType: authType
+    appSettings: {
+      AZURE_SEARCH_SERVICE: 'https://${azureCognitiveSearchName}.search.windows.net'
+      AZURE_SEARCH_INDEX: azureSearchIndex
+      AZURE_SEARCH_CONVERSATIONS_LOG_INDEX: azureSearchConversationLogIndex
+      AZURE_SEARCH_SEMANTIC_SEARCH_CONFIG: azureSearchSemanticSearchConfig
```

Comment:
I believe you are missing `AZURE_SEARCH_USE_SEMANTIC_SEARCH` value. It might be worth going through the changes that were made to deployment.bicep since you forked this as main.bicep seems to be a clean sheet of paper (I would have done the same! :-) )